amqp/transport.py

```diff
@@ -535,6 +533,9 @@ def _wrap_socket_sni(self, sock, keyfile=None, certfile=None,
         except AttributeError:
             pass  # ask forgiveness not permission
 
+        if cert_reqs is not None:
```

Comment:
If you don't mind, let's document inline what each condition is meant to do and why their order is this way and not another.
Otherwise this issue may regress.